### PR TITLE
Add series filter to search.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/17/
Test PASSed.

File: theblues/charmstore.py [hunk 1]
```diff
@@ -239,6 +239,8 @@ def search(self, text, includes=None, limit=None,
                which can be name, author, series and - in front for descending
         @param owner Optional owner. If provided, search results will only
                include entities that owner can view.
+        @param series The series to filter; can be a list of series or a
+               single serie.
```
Comment (mitechie) [inline theblues/charmstore.py]: typo, 'series'

**Comment by mitechie** [comment]
:+1: with note on typo, no QA

File: theblues/tests/test_charmstore.py [hunk 2]
```diff
@@ -421,6 +445,16 @@ def test_search_multiple_tags(self):
             results = self.cs.search('foo', tags=['databases', 'applications'])
             self.assertEqual([{'Id': 'cs:foo/bar-0'}], results)
 
+    def test_search_seriess(self):
```
Comment (jcsackett) [inline theblues/tests/test_charmstore.py]: Minor typo: series, not seriess

**Comment by jcsackett** [comment]
Code is :+1:, doing QA now.

**Comment by jcsackett** [comment]
QA notes:

```
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from theblues.charmstore import CharmStore
>>> cs = CharmStore('http://api.jujugui.org/charmstore/v4')
>>> cs.search('mysql', series='precise')
[{u'Id': u'cs:~justin-fathomdb/precise/mysql-0'}, {u'Id': u'cs:~hazmat/precise/mysql-0'}, {u'Id': u'cs:~clint-fewbar/precise/mysql-1'}, {u'Id': u'cs:~charmers/precise/cf-mysql-0'}, {u'Id': u'cs:~gandelman-a/precise/mysql-0'}, {u'Id': u'cs:~niedbalski/precise/mysql-0'}, {u'Id': u'cs:~openstack-ubuntu-testing/precise/mysql-5'}, {u'Id': u'cs:~negronjl/precise/mysql-file-permissions-1'}, {u'Id': u'cs:~justin-fathomdb/precise/mysql-proxy-0'}, {u'Id': u'cs:precise/postgresql-psql-7'}]
>>> cs.search('mysql', series='trusty')
[{u'Id': u'cs:~justin-fathomdb/trusty/mysql-1'}, {u'Id': u'cs:~cf-charmers/trusty/cf-mysql-broker-23'}, {u'Id': u'cs:~tribaal/trusty/mysql-1'}, {u'Id': u'cs:~brian-albrecht/trusty/mysql-1'}, {u'Id': u'cs:~a.rosales/trusty/mysql-0'}, {u'Id': u'cs:~bjornt/trusty/mysql-0'}, {u'Id': u'cs:~cf-charmers/trusty/mysql-0'}, {u'Id': u'cs:~prismakov/trusty/cf-mysql-broker-1'}, {u'Id': u'cs:~robru/trusty/mysql-0'}, {u'Id': u'cs:~justin-fathomdb/trusty/mysql-proxy-7'}]
>>> cs.search('mysql', series=['trusty', 'precise'])
[]
>>> exit()
```

It's worth noting that while the last call seems like an issue (I would expect all entities in either series to be shown), this is an issue in the charmstore, not this code.

https://api.jujucharms.com/charmstore/v4/search?text=mysql&series=precise,trusty

**Comment by jcsackett** [comment]
QA :ok:, thanks @fabricematrat!

File: theblues/tests/test_charmstore.py (hunk 2 above)
Comment (fabricematrat) [inline theblues/tests/test_charmstore.py]: Done.

File: theblues/charmstore.py (hunk 1 above)
Comment (fabricematrat) [inline theblues/charmstore.py]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/18/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/19/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/20/
Test PASSed.

File: RELEASE_PROCESS.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+::
+
+     git clone git@github.com:juju/theblues.git
+     cd theblues
+
+
+Find the next release number. You can check the last release by running `git
+tag` and finding the largest version number. Note that the values are sorted
+as strings, so don't just look at the last value.
```
Comment (jcsackett) [inline RELEASE_PROCESS.rst:15]: So, it's worth noting that right now we don't have a tagged version--we'll want to find the 0.0.1 sha and tag that for this to be accurate.

File: RELEASE_PROCESS.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+
+Then you can publish the release to pipy,
+
+::
+
+    python setup.py register -r pypitest
+    python setup.py sdist upload -r pypitest
+    python setup.py register -r pypi
+    python setup.py sdist upload -r pypi
```
Comment (jcsackett) [inline RELEASE_PROCESS.rst]: So these are sort of generics; what we actually use are our make targets:

```
$ make dist
$ make upload
```

If those aren't sufficient for uploading a new release, that's a bug in our makefile.

**Comment by jcsackett** [comment]
Thanks for the release doc; I've got a few notes on it, but it's mostly good to go. I suspect @mitechie might have some thoughts as well.

File: RELEASE_PROCESS.rst (hunk 4 above)
Comment (fabricematrat) [inline RELEASE_PROCESS.rst]: Done. Kool it's much better.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/21/
Test PASSed.

File: RELEASE_PROCESS.rst (hunk 3 above)
Comment (fabricematrat) [inline RELEASE_PROCESS.rst:15]: 0.0.1 Created.

**Comment by fabricematrat** [comment]
:shipit:  thanks @jaycee && @mitechie

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

**Pull request merged**